pallets/crowdloans/src/lib.rs

```diff
@@ -779,24 +795,27 @@ pub mod pallet {
                     Vaults::<T>::insert(crowdloan, vault.id, vault);
                 }
                 XcmInflightRequest::Contribute {
-                    crowdloan: index,
+                    crowdloan,
                     who,
                     amount,
                 } if !executed => {
-                    // refund
+                    let mut vault =
+                        Self::current_vault(crowdloan).ok_or(Error::<T>::VaultDoesNotExist)?;
                     T::Assets::transfer(
                         T::RelayCurrency::get(),
-                        &Self::vault_account_id(index),
+                        &Self::vault_account_id(crowdloan),
                         &who,
                         amount,
                         true,
                     )?;
+                    Self::do_update_pending(&who, &mut vault, amount, false)?;
```

Comment:
should remove pending contribution in case we can `migrate_pending` in the future but no tokens have been refunded